### ISSUE TITLE
Does crc-rs have plans to implement hardware acceleration?

### ISSUE THREAD

**Issue by RiversJin** [mrhooray/crc-rs#104]
I have noticed that the current implementation still uses table lookup method. Perhaps introducing acceleration instructions for different platforms could significantly improve the computational speed. I saw in https://github.com/mrhooray/crc-rs/issues/83 that it is currently waiting for the completion of the std::simd feature before implementing this. Is that correct?

Because I would like to implement Intel's SSE4.2 acceleration for CRC calculations in crc-rs, but I'm unsure if this aligns with crc-rs's future development plans.

**Comment by akhilles** [comment]
Yes, definitely! There were a bunch of recent changes (e.g. https://github.com/mrhooray/crc-rs/pull/76) specifically aimed at supporting multiple implementations.
